tests/unit/test_state.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Jenkins-k8s state module tests."""
+
+from unittest.mock import MagicMock
+
+import pytest
+
+import state
+
+
+@pytest.mark.parametrize(
+    "time_range",
+    [
+        pytest.param("-1-3", id="negative time range"),
```

Comment:
Should the edges be probed, i.e., -1, 0, 1, 22, 23, 24? Off by one is a common mistake that can be caught by these kind of edge tests

Reply:
I don't get your comments 
`-1` is tested here
`0`, `1`, `22`, `23` aren't invalid
`24` is tested on the next param

Reply:
I didn't mean exactly here, across the tests (both invalid and valid cases)

Reply:
Updated the test cases to include the before/edge/during/after cases :D Thanks for the review!